### PR TITLE
Add CI and unit tests for prometheus rules using promtool

### PR DESCRIPTION
**Pull request by gabrielcocenza**
- It's good to have unit tests for the prometheus alerts and also check if the prometheus_alerts.yaml is a valid file.

- using promtool is possible to check the rules and also run unit tests.

## How to test this PR

1. Install the prometheus snap:
```shell
sudo snap install prometheus
```

2. Run to check if the rules file is valid:
```shell
promtool check rules src/alert_rules/prometheus/*.yaml
```

3. Run the unit tests:
```
promtool test rules tests/unit/test_alert_rules/*.yaml
```

### CODE REVIEW

File: .github/workflows/test_prometheus_rules.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      - "**.md"
+      - "**.rst"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  promtool:
+    runs-on: ubuntu-latest
```
Comment (phvalguima) [inline .github/workflows/test_prometheus_rules.yaml:13]: Is it possible to move to self-hosted CI?

Comment (gabrielcocenza) [reply .github/workflows/test_prometheus_rules.yaml:13]: I think it is. Is there a specific reason to change to self-hosted?

**Review by zmraul — CHANGES_REQUESTED** [review]
See [comment on opensearch-dashboards](https://github.com/canonical/opensearch-dashboards-operator/pull/142#pullrequestreview-2500122619)

File: .github/workflows/test_prometheus_rules.yaml (hunk 1 above)
Comment (phvalguima) [reply .github/workflows/test_prometheus_rules.yaml:13]: Not really, just more runners. But as you removed from the pull request path, as @zmraul suggested, I am fine keeping it here.

**Review by phvalguima — APPROVED** [review]

**Review by zmraul — APPROVED** [review]

**Pull request merged**